assets/js/unminified/mode-switcher.js

```diff
@@ -11,25 +11,63 @@
 astraUpdateSwitcherAtts = function ( switchedTo ) {
 	// Update aria-label attr.
 	var modeSwticherTrigger =  document.querySelectorAll( '.ast-mode-switcher-trigger' );
-	for ( var count = 0; count < modeSwticherTrigger.length; count++ ) {
+	for ( var triggerCount = 0; triggerCount < modeSwticherTrigger.length; triggerCount++ ) {
 		if( 'dark' === switchedTo ) {
-			modeSwticherTrigger[ count ].setAttribute( 'aria-label', astraModeSwitcher.switchToLightMode );
+			modeSwticherTrigger[ triggerCount ].setAttribute( 'aria-label', astraModeSwitcher.switchToLightMode );
 		} else {
-			modeSwticherTrigger[ count ].setAttribute( 'aria-label', astraModeSwitcher.switchToDarkMode );
+			modeSwticherTrigger[ triggerCount ].setAttribute( 'aria-label', astraModeSwitcher.switchToDarkMode );
 		}
 	}
 
 	// Update title attr.
 	var modeSwticherIconButtons =  document.querySelectorAll( '.ast-mode-switcher-icon-button, .ast-mode-switcher-icon-toggle' );
-	for ( var count = 0; count < modeSwticherIconButtons.length; count++ ) {
+	for ( var buttonCount = 0; buttonCount < modeSwticherIconButtons.length; buttonCount++ ) {
 		if( 'dark' === switchedTo ) {
-			modeSwticherIconButtons[ count ].setAttribute( 'title', astraModeSwitcher.switchToLightMode );
+			modeSwticherIconButtons[ buttonCount ].setAttribute( 'title', astraModeSwitcher.switchToLightMode );
 		} else {
-			modeSwticherIconButtons[ count ].setAttribute( 'title', astraModeSwitcher.switchToDarkMode );
+			modeSwticherIconButtons[ buttonCount ].setAttribute( 'title', astraModeSwitcher.switchToDarkMode );
 		}
 	}
 }
 
+/**
+ * JS for updating toggled bubble translateX offset.
+ *
+ * Case: Due to some inconsistencies the toggled bubble look inappropriate, because we can't determine how button size (width) changes either of label font size or by icon size. That's why managing the bubble translateX offset through JS.
+ */
+astraUpdateToggleButtonSize = function() {
+	var iconWithToggle = document.querySelectorAll( '.ast-mode-switcher-icon-with-label-toggle .ast-dark-switcher-knob .ahfb-svg-iconset' );
+
+	if ( iconWithToggle.length > 0 ) {
+		var toggleAdjustCss = '',
+			styleSheet = document.createElement( 'style' );
+
+		for ( var toggleCount = 0; toggleCount < iconWithToggle.length; toggleCount++ ) {
+			var parentOffset = iconWithToggle[toggleCount].closest( '.ast-mode-switcher-trigger' );
```

Comment:
```suggestion
			var parentOffset = iconWithToggle[toggleCount].closest( '.ast-mode-switcher-trigger' );
```
@imnavanath Instead of closest can we use - parents or child function?

Reply:
@premanshup updated with using - `offsetParent` it worked. But curious why closest don't use? Review commit - https://github.com/brainstormforce/astra/pull/3856/commits/6f6c00606b9dea102056d2c0573a7b4c7aa30189

Reply:
@imnavanath Using parent or child will make sure that we are searching the current div/markup structure. If in future we change some structure and another elements CSS class comes closer than its own structure then it will target that CSS class instead of current markups CSS Class.